FIRE-35019 Add LLHUBNameTag background to floating text and hover highlights

minerjr opened the pull request:
Added new feature for allowing backgrounds behind floating prim text as well as supporting highlighting the text by changing render order as well as disabling transparency. User is able to enable/disable feature in preferences and supports a few options.

## Firestorm Pull Request Checklist
Thank you for contributing to the Phoenix Firestorm Project.
We will endeavour to review you changes and accept/reject/request changes as soon as possible. 
Please read and follow the [Firestorm Pull Request Guidelines](https://github.com/firestormviewer/phoenix-firestorm/blob/master/FS_PR_GUIDELINES.md) to reduce the likelihood that we need to ask for "Bureaucratic" changes to make the code comply with our workflows.

beqjanus commented:
Hi @minerjr 
Thanks for this. I've only skimmed through so far and it looks reasonable, though I'm not clear yet what it looks like. There appear to be some anomalies in the source markup. 

In general, if you change a single line then it is ok to suffix that with "<FS:minerjr/> reason for change" but for larger changes it is important that the existing code that has been altered is preserved. This ensures that when @Ansariel merges in the upstream code any conflicts are idenitifable and she can comapre the changes that were made upstream with the commented code blocks and make an educated guess at whether more work is needed before we release it. In most places you've done a pretty good job of marking up correctly but there are some more complicated sections where the markers are not preserving the orginal code. 

Let me know if that makes sense and if you need I can try to list the ones I've spotted, but it might be easier for you to review and update them yourself, given that you know the changes.

minerjr commented:
Thanks for the feedback. Sure, that makes sense. I only modified a few files, so I will take the original master, refork it and and port over my changes again, but preserve the alterations in commented out code and put my changes below. 

Also, should I mark up the commented out sections to indicate that they were commented out for a given JIRA issue number?

I attached some screen shots showing the 2 different scenarios in the JIRA issue. But I will add them to this post.
The Issue 01 and 02 show what the current state of the viewer is. The other show off some of 2 of the possible configurations, only highlighted and all prims. 

![SL_all_prims](https://github.com/user-attachments/assets/fad93542-2f9a-4523-8778-1cb0ff64cd05)
![SL_all_prims_camera_far](https://github.com/user-attachments/assets/0e4a5640-c637-4940-8fda-fd7a9d6b4adf)
![SL_Issue01](https://github.com/user-attachments/assets/69e8ce5b-f187-41a6-bf42-36c830a4684b)
![SL_issue02](https://github.com/user-attachments/assets/b433307d-2acf-4410-8391-fc7e35819213)
![SL_only_highlighted_prim](https://github.com/user-attachments/assets/90f1780c-40d1-4550-9c1f-27e34913548e)
![SL_only_highlighted_prim2](https://github.com/user-attachments/assets/34fe8fce-d9f7-4cf8-9ed2-a18c29a134b0)

minerjr commented:
Hi @beqjanus I was wondering if you could look at my new re-factored version before I create a pull, so I can maybe clean it up any more then I already have. 

You should be able to view the commit at: [https://github.com/minerjr/phoenix-firestorm/commit/2fbac952c55deaa77428e1cc82a4222dc2121e52](https://github.com/minerjr/phoenix-firestorm/commit/2fbac952c55deaa77428e1cc82a4222dc2121e52)

I was able to clean up the code and applied a few new learnings to reduce the amount of code change as well as make sure if I did modify code, it was commented out and included in between the tags. I also added comments in the XML files that were modified and added an additional user control to change the background opacity.

I am going to put the following images into the pull as well.

Current state of SL with DFS farm objects for example - Very hard to read most of the text, lots of overlap
![FIRE-35019-01](https://github.com/user-attachments/assets/3878edcf-94a4-40cf-8ca5-cc9272408454)

New Floating Text Hover Highlight feature - hover over far away prim makes it text appear over all other text, even fully faded text.
![FIRE-35019-02](https://github.com/user-attachments/assets/eff7ba6a-c43a-4469-a725-ca94adee0298)

New Floating Text Show Background feature - now when hovering over a prim, a background appears behind the text, to make it readable.
![FIRE-35019-03](https://github.com/user-attachments/assets/0dc6bbac-fbab-40a7-8521-4a20c5c05f05)

Also support having all prims have a background appear on floating text.
![FIRE-35019-05](https://github.com/user-attachments/assets/da2607c6-ec57-45f7-92cf-fbed5c01f42b)

Can also highlight a single prim, and its text box will become fully opaque and again drawn over top of the other text.
![FIRE-35019-06](https://github.com/user-attachments/assets/f513bf54-3f50-4143-816c-3021f60d864f)

User can also change the opacity of the the floating text background. Default 75% shown
![FIRE-35019-07](https://github.com/user-attachments/assets/d5e43d17-078f-4e10-a77f-51189ac59729)

Example at 40% opacity.
![FIRE-35019-08](https://github.com/user-attachments/assets/810397dd-64e1-433e-b05f-224b214d7d05)

Again thanks for the feedback and I hope that I can continue to contribute to this project. I am next going to tackle another change I had created, but I will follow the same re-factor steps I did for this one to get it into better shape as well.
